bigquery_etl/glam/templates/clients_histogram_bucket_counts_v1.sql

```diff
@@ -0,0 +1,154 @@
+CREATE TEMP FUNCTION udf_normalized_sum(arrs ARRAY<STRUCT<key STRING, value INT64>>)
+RETURNS ARRAY<STRUCT<key STRING, value FLOAT64>> AS (
+  -- Returns the normalized sum of the input maps.
+  -- It returns the total_count[k] / SUM(total_count)
+  -- for each key k.
+  (
+    WITH total_counts AS (
+      SELECT
+        sum(a.value) AS total_count
+      FROM
+        UNNEST(arrs) AS a
+    ),
+    summed_counts AS (
+      SELECT
+        a.key AS k,
+        SUM(a.value) AS v
+      FROM
+        UNNEST(arrs) AS a
+      GROUP BY
+        a.key
+    ),
+    final_values AS (
+      SELECT
+        STRUCT<key STRING, value FLOAT64>(
+          k,
+          COALESCE(SAFE_DIVIDE(1.0 * v, total_count), 0)
+        ) AS record
+      FROM
+        summed_counts
+      CROSS JOIN
+        total_counts
+    )
+    SELECT
+      ARRAY_AGG(record)
+    FROM
+      final_values
+  )
+);
+
+CREATE TEMP FUNCTION udf_normalize_histograms(
+  arrs ARRAY<
+    STRUCT<
+      latest_version INT64,
+      metric STRING,
+      metric_type STRING,
+      key STRING,
+      agg_type STRING,
+      sum INT64,
+      value ARRAY<STRUCT<key STRING, value INT64>>
+    >
+  >
+)
+RETURNS ARRAY<
+  STRUCT<
+    latest_version INT64,
+    metric STRING,
+    metric_type STRING,
+    key STRING,
+    agg_type STRING,
+    aggregates ARRAY<STRUCT<key STRING, value FLOAT64>>
+  >
+> AS (
+  (
+    WITH normalized AS (
+      SELECT
+        {{ metric_attributes }},
+        -- NOTE: dropping the actual sum here, since it isn't being used
```

Comment:
Should the actual sums of the histograms be surfaced somewhere, or dropped entirely from the aggregates pipeline?

Reply:
I think using the actual sum would cause the sum of the normalized values to be less than 1, because the recorded values are floored to the nearest bucket.

Reply:
ok lets not use it then.

Reply:
I've dropped the sums from all of the queries.